core/src/compaction/validator.rs

```diff
@@ -0,0 +1,169 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Contains file writer API, and provides methods to write row groups and columns by
+//! using row group writers and column writers respectively.
+
+use std::collections::HashSet;
+use std::sync::Arc;
+
+use futures::StreamExt;
+use iceberg::spec::{DataFile, Schema};
+use iceberg::table::Table;
+
+use crate::error::Result;
+use crate::executor::InputFileScanTasks;
+use crate::executor::datafusion::datafusion_processor::{
+    DataFusionTaskContext, DatafusionProcessor,
+};
+use crate::{CompactionConfig, CompactionError};
+
+pub struct CompactionValidator {
+    datafusion_processor: DatafusionProcessor,
+    input_datafusion_task_ctx: Option<DataFusionTaskContext>,
+    output_datafusion_task_ctx: Option<DataFusionTaskContext>,
+    table_ident: String,
+    catalog_name: String,
+}
+
+impl CompactionValidator {
```

Comment:
I think we can just validate the input and output files' row counts instead of comparing two files row by row, since sorting a large table might be too expensive.